### PR TITLE
Solution Week-09: MCC = 0.6481979128125026

### PR DESCRIPTION
**Pull request by DrKeks1010**
With a decision tree, i couldn't improve over an MCC of 0.6551476590190098 in the training csv, which produzed a MCC of 0.6481979128125026 for the test csv

### CODE REVIEW

File: Exercises/JonasPutz/exercises_09.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn import tree
+from sklearn.metrics import matthews_corrcoef
+from sklearn.model_selection import cross_val_score
+
+def prepare_data(path: Path) -> pd.DataFrame:
+    """ Function to prepare the dataframe from an input path"""
+
+    dataframe = pd.read_csv(path)
```
Comment (TrueEsel) [inline Exercises/JonasPutz/exercises_09.py:30]: didnt know that was possible

File: Exercises/JonasPutz/exercises_09.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+x = train_df.drop(columns = "Survived")
+y = train_df[["Survived"]]
+x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, stratify = y)
+
+#finding best max depth
+mean_scores = []
+for depth in [2, 3, 4, 5, 6, 7, 8, 9, 10, 30]:
+    classifier = tree.DecisionTreeClassifier(max_depth=depth)
+    scores = cross_val_score(classifier, x_train, y_train, cv=3, scoring='matthews_corrcoef')
+    mean_scores.append((depth, scores.mean(), scores.std()))
```
Comment (TrueEsel) [inline Exercises/JonasPutz/exercises_09.py:58]: I think the same thing could be achived using GridSearchCV

**Review by TrueEsel — APPROVED** [review]
Code is looking good. I believe that finding the best depth could have been done easier but your way is working so I dont consider that to be a huge issue.

**Comment by fabischw** [comment]
I will be merging this PR if it does not get merged by the PR creator within the next 3 days in an effort to clean up the repo (More information can be found in [Issue #228](https://github.com/dhbw-ma-ppp/ppp-2023/issues/228))

**Pull request merged**